OpenSim/Simulation/SimbodyEngine/Joint.h

```diff
@@ -568,6 +574,17 @@ inline int Joint::getNumMobilities(const SimTK::MobilizedBody::Free& mobod) cons
     return 6;
 }
 
+class EmptyCoordinateSet : public Exception {
```

Comment:
Try making the Exception type agnostic to the underlying data structure containing the Coordinates, e.g. `HasNoCoordinates`. If you change to property list of Coordinates do you want to have to change the Exception?